### PR TITLE
ci: enable yamllint [INFENG-116]

### PR DESCRIPTION
**Pull request by dannysauer**
## Description

Add yamllint check for .github/** only

Reformat just about all of the files to include yaml document begin headers, use wrapping blocks for text which is fine to wrap, add the special comment for the `on:` line in actions files (sigh), and reindent a couple of things which weren't "properly" indented.

## Test Plan

The two golang actions which had an actual change (`on` variable value was quoted in lint-bindings and lint-go) will run as part of the PR.  The other change was reformatting some text in issue templates; will have to create new issues to make sure they still look the same afterwards (they *should*, assuming I understand yaml multiline values correctly).

## Checklist
- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: .github/workflows/test-cli.yml [hunk 1]
```diff
@@ -35,13 +35,13 @@ jobs:
           python -m pip install --upgrade pip
           pip --version
           # Windows needs special treatment of cryptography for unknown reasons.
-          # LMDB also has special requirements that only apply to windows (patch-ng).
```
Comment (loksonarius) [inline .github/workflows/test-cli.yml:38]: Feels arbitrary but sure

**Review by loksonarius — APPROVED** [review]
Please double check out issue and other templates after merging to make sure the description renders as expected. Everything else seemed fine

File: .github/workflows/test-cli.yml (hunk 1 above)
Comment (dannysauer) [reply .github/workflows/test-cli.yml:38]: That was just to get it under 80 chars. 😂

**Comment by dannysauer** [comment]
Validated issue templates still work properly.  Hooray for knowing YAML! :p

**Pull request merged**